src/dispatch/plugins/dispatch_slack/views.py

```diff
@@ -681,6 +691,96 @@ def get_channel_id(event_body: dict):
+        channel_id=channel_id,
+        incident_types=type_options,
+        incident_priorities=priority_options
+    )
+
+    dispatch_slack_service.open_view_for_user(client=slack_client, trigger_id=trigger_id, view=modal_view_template)
+
+
+def parse_submitted_form(view_data: dict):
+    """Parse the submitted data and return important / required fields for dispatch to start an incident"""
+    parsed_data = {}
+    state_elem = view_data.get('state')
+    state_values = state_elem.get('values')
+    for state in state_values:
+        state_key_value_pair = state_values[state]
+
+        for elem_key in state_key_value_pair:
+            elem_key_value_pair = state_values[state][elem_key]
+
+            if elem_key_value_pair.get('selected_option') and elem_key_value_pair.get('selected_option').get('value'):
+                parsed_data[state] = {
+                    'name': elem_key_value_pair.get('selected_option').get('text').get('text'),
+                    'value': elem_key_value_pair.get('selected_option').get('value')
+                }
+            else:
+                parsed_data[state] = elem_key_value_pair.get('value')
+
+    return parsed_data
+
+
+def create_incident_from_submitted_form(
+        user_id: str, user_email: str, incident_id: int, action: dict, db_session=None
+):
+    from dispatch.incident.service import create
+    from dispatch.incident.flows import incident_create_flow
+    from dispatch.incident.enums import IncidentSlackViewBlockId, IncidentStatus
+
+    submitted_form = action.get('view')
+
+    # Fetch channel_id from the callback_id submitted when the modal was opened
+    channel_id = submitted_form.get('callback_id').split('__')[1]
+
+    parsed_form_data = parse_submitted_form(submitted_form)
+
+    requested_form_title = parsed_form_data.get(IncidentSlackViewBlockId.title)
+    requested_form_description = parsed_form_data.get(IncidentSlackViewBlockId.description)
+    requested_form_incident_type = parsed_form_data.get(IncidentSlackViewBlockId.type)
+    requested_form_incident_priority = parsed_form_data.get(IncidentSlackViewBlockId.priority)
+
+    # create the incident object
+    incident = create(
+        db_session=db_session,
+        title=requested_form_title,
+        status=IncidentStatus.active,
+        description=requested_form_description,
+        incident_type=requested_form_incident_type,
+        incident_priority=requested_form_incident_priority,
+        reporter_email=user_email
+    )
+
+    incident_create_flow(incident_id=incident.id)
+
+    # respond back to user with a message that (a user has started an incident)
+    msg_template = create_incident_confirmation_msg(
+        user=f'<@{user_id}>',
+        title=requested_form_title,
+        priority=requested_form_incident_priority,
+        t=requested_form_incident_type,
+    )
+
+    dispatch_slack_service.send_message(client=slack_client, conversation_id=channel_id, text=msg_template)
```

Comment:
I think this should be an ephemeral message to the user. Dispatch already notifies about new reported incidents in the channels defined [here](https://hawkins.gitbook.io/dispatch/configuration/app#incident_notification_conversations-default).